tests/_test_process_executor.py

```diff
@@ -79,24 +78,32 @@ def test_run_after_shutdown(self):
         with pytest.raises(RuntimeError):
             self.executor.submit(pow, 2, 5)
 
-    @pytest.mark.skipif(sys.version_info < (3, 4),
-                        reason="requires python>3.4")
     def test_interpreter_shutdown(self):
-        from .script_helper import assert_python_ok
-        # Test the atexit hook for shutdown of worker threads and processes
-        rc, out, err = assert_python_ok('-c', """if 1:
-            from loky.process_executor import {executor_type}
-            from time import sleep
-            from tests._test_process_executor import sleep_and_print
-            t = {executor_type}(4)
-            t.submit(id, 42).result()
-            t.map(sleep_and_print, [0.1] * 10, range(10))
-            """.format(executor_type=self.executor_type.__name__))
-        # Errors in atexit hooks don't change the process exit code, check
-        # stderr manually.
-        assert not err
-        unique_out_bytes = set(out.replace(b"\r\n", b"").replace(b"\n", b""))
-        assert len(unique_out_bytes) == 10
+        n_jobs = 4
+        code = "\n".join([
+            'from loky.process_executor import {executor_type}',
+            'from loky.backend import get_context',
+            'from time import sleep',
+            'from tests._test_process_executor import sleep_and_print',
+            'context = get_context("{start_method}")',
+            't = {executor_type}({n_jobs}, context=context)',
+            't.submit(id, 42).result()',
+            't.map(sleep_and_print, [0.1] * 2 * {n_jobs}, range(2 * {n_jobs}))'
+        ]).format(executor_type=self.executor_type.__name__,
+                  start_method=self.context.get_start_method(), n_jobs=n_jobs)
```

Comment:
This is done with the `-c` option here and this is why it works. It would not work without the safeguard here if this was a script. I do not think it is useful to test it from a file but I can do it if needed.